charts/k8s-watcher/templates/clusterrole.yaml

```diff
@@ -423,6 +423,23 @@ rules:
       - get
       - watch
 {{- end}}
+
+{{- if .Values.watcher.resources.allowReadAll }}
+  - apiGroups: ["*"]
+    resources: ["*"]
+    verbs:     ["get", "watch", "list"]
+{{- end}}
+
+{{- $numOfCustomApiGroups := len .Values.watcher.resources.customAPIGroups }}
+{{- $shouldLimitApiGroups := gt $numOfCustomApiGroups 0 }}
+{{- if and $shouldLimitApiGroups (not .Values.watcher.resources.allowReadAll) }}
+  {{- range $cag := .Values.watcher.resources.customAPIGroups }}
+  - apiGroups: {{ $cag.apiGroups }}
+  - resources: {{ $cag.resources }}
+  - verbs: {{ $cag.verbs }}
```

Comment:
I think it is not the correct syntax to have apiGroups resources and verb in an array
it should be
```yaml
- apiGroups: {{ $cag.apiGroups }}
  resources: {{ $cag.resources }}
  verbs: {{ $cag.verbs }}
```

also, I think it is enough to have only the api groups in the `customAPIGroups` and use * for resources and get list watch as verbs

Reply:
Thanks for the correction. I am not convinced that we might need to remove the resources/verbs with this current configuration name. Now the name is `customAPIGroups` and not `customReadAPIGroups` so it could confuse that it will be about getting/list/watch, and will set all the resources. WDYT?

Reply:
as I see it we have two options:
1. allow users to list custom api groups (we could name it customReadAPIGroups) and we will add the groups with read permissions on all resources under these groups
2. allow users to freely add rules to the agent RBAC policy - groups, resources and verbs. in that case I would just call the value additionalRBACPermissions and embed the provided YAML as-is into the agent's ClusterRole

Reply:
Fixed.